### PR TITLE
For non-convex quad fix centroid calculation to avoid unrealistic z values

### PR DESCRIPTION
**Pull request by JasonGlazer**
Addresses #5497

This is the second pull request related to this fix. The first got messed up so this is a redo.
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [x] If any virtual classes, ensure virtual destructor included, other things

### CODE REVIEW

**Comment by JasonGlazer** [comment]
The diffs in the CI are all on the 5ZoneGeometryTransform.idf file. There were issues related to this file found and posted to #5849 during the development of the fix. In addition, the centroid for the areas for some of the surfaces in the file were being miscalculated originally. Since the centroid calculation was fixed, this impacted the surfaces and flowed down to the zone and HVAC impacts. 

@Myoldmopar and @mjwitte I believe this is ready to be reviewed and merged.

File: src/EnergyPlus/SurfaceGeometry.cc [hunk 1]
```diff
@@ -10052,16 +10052,36 @@ namespace SurfaceGeometry {
...
 				}
 
 				// get area fraction of triangles.
-				Real64 const Tri1Area( AreaPolygon( 3, Triangle1 ) / TotalArea );
-				Real64 const Tri2Area( AreaPolygon( 3, Triangle2 ) / TotalArea );
+				Real64 Tri1Area( AreaPolygon( 3, Triangle1 ) / TotalArea );
+				Real64 Tri2Area( AreaPolygon( 3, Triangle2 ) / TotalArea );
+
+				// check if sum of fractions are slightly greater than 1.0 which is a symptom of the triangles for a non-convex quadralateral using the wrong two triangles
+				if ( (Tri1Area + Tri2Area) > 1.05 ) {
```
Comment (mjwitte) [inline src/EnergyPlus/SurfaceGeometry.cc:10067]: @JasonGlazer What's the basis for >1.05 as the test?  Could a bad surface slip through with, say 1.04?

**Review by mjwitte — COMMENTED** [review]
Tested with the defect file before/after.  Couldn't find a way to report surface centroids?  So added reporting of outside surface wind speed and temperatures to verify that other surfaces keep the same values and that the offending surface has a reasonable value. That much all looks good.

File: src/EnergyPlus/SurfaceGeometry.cc (hunk 1 above)
Comment (JasonGlazer) [reply src/EnergyPlus/SurfaceGeometry.cc:10067]: I tried 1.0 and that tripped up a lot of surfaces that were ok so I used engineering judgment to put it at 1.05. I did not do any kind of analysis to justify that number.

**Review by mjwitte — APPROVED** [review]

**Pull request merged**